### PR TITLE
Remove some non-vectorized code

### PR DESCRIPTION
**Pull request by decster**

### CODE REVIEW

File: be/src/exec/parquet/encoding_dict.h [hunk 1]
```diff
@@ -97,6 +97,7 @@ class DictDecoder final : public Decoder {
     }
 
     Status next_batch(size_t count, ColumnContentType content_type, vectorized::Column* dst) override {
+        _indexes.reserve(count);
```
Comment (trueeyu) [inline be/src/exec/parquet/encoding_dict.h:100]: _indexes has already resizd in set_dict(), why reserve here

**Review by chaoyli — APPROVED** [review]

File: be/src/exec/parquet/encoding_dict.h (hunk 1 above)
Comment (decster) [reply be/src/exec/parquet/encoding_dict.h:100]: `count` may be greater than `config::vector_chunk_size`, that why some UT failed

Comment (trueeyu) [reply be/src/exec/parquet/encoding_dict.h:100]: In actual use, it is impossible to be less than

Comment (trueeyu) [reply be/src/exec/parquet/encoding_dict.h:100]: no need to add this reserve just for single test

Comment (decster) [reply be/src/exec/parquet/encoding_dict.h:100]: > In actual use, it is impossible to be less than
There is no actual code to enforce that, right?

Comment (decster) [reply be/src/exec/parquet/encoding_dict.h:100]: > no need to add this reserve just for single test

That's for preventing future potential bugs, not just for this single test

Comment (trueeyu) [reply be/src/exec/parquet/encoding_dict.h:100]: https://github.com/StarRocks/starrocks/blob/0ab1a342f4d25c6212e0de2eadc7360c72aeeca7/be/src/exec/parquet/file_reader.cpp#L431

Comment (trueeyu) [reply be/src/exec/parquet/encoding_dict.h:100]: The chunk generated by the storage engine should not be greater than chunk size

Comment (decster) [reply be/src/exec/parquet/encoding_dict.h:100]: This is file_reader, not decoder

**Review by trueeyu — APPROVED** [review]

File: be/src/exec/parquet/encoding_dict.h (hunk 1 above)
Comment (trueeyu) [reply be/src/exec/parquet/encoding_dict.h:100]: file_reader -> group_reader -> page_reader -> decoder all use the  "n"

**Pull request merged**